### PR TITLE
Show Blaze entry points behind a feature flag

### PR DESCRIPTION
**Pull request by jaclync**
Part of #9866 

## Why

For the initial development, we can use a local feature flag to build the UI before we implement the more complex eligibility check. For simpler review and testing, this PR just contains the local feature flag (debug/alpha builds only) and the two entry points behind the feature flags. The Blaze flow will be implemented in a future PR.

## Description

A feature flag `blaze` was created. It's used for these two entry points:

- Menu > General: under the Payments row
- Products > Product > More: under the Share action

### Menu tab

- Log in to a store if needed
- Go to the Menu tab --> there should be a new row "Blaze" under the Payments row in the General section

### Product menu

- Go to the Products tab
- Create a product if the store has no products
- Tap on a product
- Tap on the ellipsis menu in the navigation bar --> there should be a new action "Promote with Blaze" under the Share action

---

- [x] @jaclync tests that the Blaze entry points are not shown when the feature flag is false

## Screenshots

menu tab | product more menu
-- | --
Simulator Screen Shot - iPhone 14 Pro - 2023-06-05 at 14 40 48 | Simulator Screen Shot - iPhone 14 Pro - 2023-06-05 at 09 24 21

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift [hunk 1]
```diff
@@ -1729,6 +1735,7 @@ private enum ActionSheetStrings {
     static let viewProduct = NSLocalizedString("View Product in Store",
                                                comment: "Button title View product in store in Edit Product More Options Action Sheet")
     static let share = NSLocalizedString("Share", comment: "Button title Share in Edit Product More Options Action Sheet")
+    static let promoteWithBlaze = NSLocalizedString("Promote with Blaze", comment: "Button title Share in Edit Product More Options Action Sheet")
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift]: Nit: I think the comment needs to be updated.

File: WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift [hunk 2]
```diff
@@ -138,6 +138,14 @@ final class HubMenuViewModel: ObservableObject {
             }
         }
 
+        // Blaze menu.
+        // TODO: 9866 - async eligibility check
+        if featureFlagService.isFeatureFlagEnabled(.blaze), let index = generalElements.firstIndex(where: { item in
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift]: Unless we want to skip adding the Blaze row if the payments row is not present, I think we should fall back to appending the Blaze row.

**Review by selanthiraiyan — APPROVED** [review]
:shipit:

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift (hunk 1 above)
Comment (jaclync) [reply WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift]: Fixed in https://github.com/woocommerce/woocommerce-ios/pull/9868/commits/dc76e4c2a64ae852f30b4fd4f41e0b7fa3c55c98

File: WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift (hunk 2 above)
Comment (jaclync) [reply WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift]: Right now the Payments row is always visible, but I see your point we can append the row when the Payments row is conditionally hidden somehow in https://github.com/woocommerce/woocommerce-ios/pull/9868/commits/a36ba76f863a4c92b162e0212b669037f6bd3200. Also added some test cases for the menu elements to ensure the placement of the Blaze row in https://github.com/woocommerce/woocommerce-ios/pull/9868/commits/178d912606c6051709cbda487461406e2ba445d7.

**Pull request merged**